src/Analyzers/CSharp/Analysis/RemoveUnnecessaryElseAnalyzer.cs

```diff
@@ -61,22 +58,19 @@ private static bool IsFixable(ElseClauseSyntax elseClause, SemanticModel semanti
         if (ifStatementStatement is not BlockSyntax ifBlock)
             return CSharpFacts.IsJumpStatement(ifStatementStatement.Kind());
 
-        if (elseClause.Statement is BlockSyntax elseBlock)
```

Comment:
These diffs do not change behaviour. They are just reducing nesting / Moving the checks that require a semantic model later.